halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/DeploymentEnvironmentService.java

```diff
@@ -56,14 +62,26 @@ public DeploymentEnvironment getDeploymentEnvironment(String deploymentName) {
 
   public void setDeploymentEnvironment(String deploymentName, DeploymentEnvironment newDeploymentEnvironment) {
     DeploymentConfiguration deploymentConfiguration = deploymentService.getDeploymentConfiguration(deploymentName);
+
+    String defaultLocation = lookupDefaultLocation(deploymentName, newDeploymentEnvironment.getAccountName());
+    String location = newDeploymentEnvironment.getLocation();
+    newDeploymentEnvironment.setLocation(StringUtils.isEmpty(location) ? defaultLocation : location);
```

Comment:
I don't think this is the right place to set this. Once a user first chooses a provider they will be locked into that providers choice of location until they manually set it. I'd prefer that the location doesn't have to be modified by the user unless they explicitly start making edits to it. 

Imagine a user selects a K8S account to deploy to, immediately changes their mind and switches to a GCE account. The deploy would fail for the weird reason that there is no "spinnaker" zone. On the other hand, if they'd manually edited the location to be something like "spinnaker-prod", and then changed to a GCE account and saw that "spinnaker-prod" is not an available zone they'd at least understand where that failure is coming from, and what they edited to cause it.